sbi/inference/posteriors/mcmc_posterior.py

```diff
@@ -237,7 +237,7 @@ def sample(
         self.potential_ = self._prepare_potential(method)  # type: ignore
 
         initial_params = self._get_initial_params(
-            init_strategy, num_chains, num_workers, show_progress_bars  # type: ignore
+            init_strategy, num_chains, num_workers, show_progress_bars, **dict(sir_batch_size=init_strategy_num_candidates) # type: ignore
```

Comment:
Hm I think this is confusing. With this implementation, `sir` will draw `sir_batch_size * sir_num_batches` samples, no? `sir_num_batches` [is 10](https://github.com/mackelab/sbi/blob/ce389f8ce633985b8ea7cf96b1ea0001daac3dd5/sbi/samplers/mcmc/init_strategy.py#L74), so it will draw 10 times more samples than requested. I would be fine with setting the default of `sir_num_batches` to `1` for both `sir` and `resample`. This should solve the problem

Reply:
I agree. Given that the default `num_candidates` is only 1000 it should be fine to draw them in one batch per default. However, at the moment we are still hiding `sir_batch_size` and `sir_bum_batches` from the user. A solution would be to add something like `init_strategy_parameters` as kwarg to `MCMCPosterior` and to `sample()`. 
Would you have objections to that?

Reply:
We would then have `init_strategy_num_candidates`, `sir_batch_size`, and `sir_num_batches`, right?

I think there would be some redundancy, so not sure if this is an optimal solution.

I'm personally fine with `sir_num_batches` not being accessible for the user for now. Usually people will use fewer than 1 million `init_strategy_parameters` and thus a single batch should be fine.

Reply:
Also note that when using the the importance sampler and `sir_num_batches>1`, we still run only a single batch of `sir_num_batches*sir_batch_size`, see [here](https://github.com/mackelab/sbi/blob/ce389f8ce633985b8ea7cf96b1ea0001daac3dd5/sbi/samplers/mcmc/init_strategy.py#L63)

Reply:
I would also be fine with deprecating `init_strategy_num_candidates` and instead introducing `init_strategy_parameters` as you suggested.

Reply:
I was confused by the variable names and refactored--hope you agree! 

Yes, I think in the long run this will give us more flexibility. I will use that.

Reply:
Done, and ready for another review. Thanks!